e2e/test/provisioning/ProvisioningServiceClientE2ETests.cs

```diff
@@ -110,6 +111,45 @@ public async Task ProvisioningServiceClient_GetEnrollmentGroupAttestation_Symmet
             await ProvisioningServiceClient_GetEnrollmentGroupAttestation(AttestationMechanismType.SymmetricKey);
         }
 
+        [LoggedTestMethod]
+        public async Task ProvisioningServiceClient_TokenCredentialAuth_Success()
+        {
+            using var provisioningServiceClient = ProvisioningServiceClient.Create(
+                TestConfiguration.Provisioning.GetProvisioningHostName(),
+                TestConfiguration.Provisioning.GetClientSecretCredential());
+
+            IndividualEnrollment individualEnrollment = await CreateIndividualEnrollment(provisioningServiceClient, AttestationMechanismType.SymmetricKey, null, AllocationPolicy.Static, null, null, null);
+
+            //act
+            IndividualEnrollment individualEnrollmentResult = await provisioningServiceClient.CreateOrUpdateIndividualEnrollmentAsync(individualEnrollment);
+
+            //assert
+            Assert.IsNotNull(individualEnrollmentResult);
```

Comment:
Prefer FluentAssertions.

```suggestion
            individualEnrollmentResult.Should().NotBeNull();
```